test/e2e/serial/tests/tolerations.go

```diff
@@ -350,6 +351,83 @@ var _ = Describe("[serial][disruptive][slow][rtetols] numaresources RTE tolerati
 					g.Expect(int(updatedDs.Status.NumberReady)).To(Equal(len(pods)), "updated DS ready=%v expected pods", updatedDs.Status.NumberReady, len(pods))
 				}).WithTimeout(5 * time.Minute).WithPolling(10 * time.Second).Should(Succeed())
 			})
+
+			It("[tier3][test_id:72859] should not restart a running RTE pod on tainted node with NoSchedule effect", func(ctx context.Context) {
+				By("taint one worker node")
+				var err error
+				workers, err = nodes.GetWorkerNodes(fxt.Client, ctx)
+				Expect(err).ToNot(HaveOccurred())
+
+				tnts, _, err = taints.ParseTaints([]string{testTaint()})
+				Expect(err).ToNot(HaveOccurred())
+
+				tnt := &tnts[0]
+				taintedNode := &workers[0]
+				Eventually(func() error {
+					var err error
+					node := &corev1.Node{}
+					err = fxt.Client.Get(ctx, client.ObjectKeyFromObject(taintedNode), node)
+					if err != nil {
+						return err
+					}
+
+					updatedNode, updated, err := taints.AddOrUpdateTaint(node, tnt)
+					if err != nil {
+						return err
+					}
+					if !updated {
+						return nil
+					}
+
+					klog.Infof("adding taint: %q to node: %q", tnt.String(), updatedNode.Name)
+					err = fxt.Client.Update(ctx, updatedNode)
+					if err != nil {
+						return err
+					}
+					return nil
+				}).WithPolling(1 * time.Second).WithTimeout(1 * time.Minute).ShouldNot(HaveOccurred())
+				targetNodeNames = append(targetNodeNames, taintedNode.Name)
+				klog.Infof("considering node: %q tainted with %q", taintedNode.Name, tnt.String())
+
+				By("trigger an RTE pod restart on the tainted node by deleting the pod")
+				ds := appsv1.DaemonSet{}
+				err = fxt.Client.Get(ctx, client.ObjectKey(dsKey), &ds)
+				Expect(err).ToNot(HaveOccurred())
+
+				klog.Info("verify RTE pods before triggering the restart still include the pod on the tainted node")
+				pods, err := podlist.With(fxt.Client).ByDaemonset(ctx, ds)
+				Expect(err).NotTo(HaveOccurred(), "Unable to get pods from daemonset %q: %v", ds.Name, err)
+				Expect(len(pods)).To(Equal(len(workers)), "pods number is not as expected for RTE daemonset: expected %d found %d", len(workers), ds.Status.DesiredNumberScheduled)
+
+				var podToDelete corev1.Pod
+				for _, pod := range pods {
+					if pod.Spec.NodeName == taintedNode.Name {
+						podToDelete = pod
+						break
+					}
+				}
+				Expect(podToDelete).NotTo(BeNil())
```

Comment:
this is an object (not a pointer) so can it ever be nil?

Reply:
this is perhaps the biggest/only actual defect I found so far in this PR

Reply:
indeed the object won't be nil. changed the check to verify the pod name isn't empty. Thank you for catching this.